state/factory/util.go

```diff
@@ -96,6 +97,28 @@ func generateWorkingSetCacheKey(blkHeader block.Header, producerAddr string) has
 	return hash.Hash256b(sum)
 }
 
+func getWorkingSetByHeight(c cache.LRUCache, h uint64) *workingSet {
+	if h == 0 {
+		return nil
+	}
+	if data, ok := c.Get(h); ok {
+		if ws, ok := data.(*workingSet); ok {
+			return ws
+		}
+	}
+	return nil
+}
+
+func abandonWorkingSets(c cache.LRUCache, h uint64) {
+	for ; ; h++ {
+		if ws := getWorkingSetByHeight(c, h); ws != nil {
+			c.Remove(h)
+		} else {
+			break
+		}
+	}
+}
```

Comment:
both factory and statedb have the same `workingset cache.LRUCache`, these can become a new small component `WorkingSetCollection` later